src/Pact/Analyze/Eval/Term.hs

```diff
@@ -549,30 +549,33 @@ evalTerm = \case
     rowWriteCount tn sRk += 1
     tagAccessKey mtWrites tid sRk writeSucceeds
 
-    Just rowETy <- view $ aeTableSchemas.at tn
-    prevAVals <- case rowETy of
-      EType rowTy@(SObject _) -> do
-        rowAVals <- peekFields tn sRk rowTy
-        -- assume invariants for values already in the DB:
-        applyInvariants tn rowAVals $ mapM_ addConstraint
-        pure rowAVals
-      _ -> throwErrorNoLoc $ FailureMessage $ "evalTerm: Write: Unrecognized type in row position: " <> T.pack (show rowETy)
-
-    writeFields writeType tid tn sRk obj objTy
-
-    let newAVals = aValsOfObj schema (_sSbv obj)
-        -- NOTE: left-biased union prefers the new values:
-        nextAVals = Map.union newAVals prevAVals
-
-    applyInvariants tn nextAVals $ \invariants' ->
-      let fs :: ZipList (Located (SBV Bool) -> Located (SBV Bool))
-          fs = ZipList $ (\s -> fmap (_sSbv s .&&)) <$> invariants'
-      in maintainsInvariants . at tn . _Just %= (fs <*>)
-
-    --
-    -- TODO: make a constant on the pact side that this uses:
-    --
-    pure $ literalS "Write succeeded"
+    -- Just rowETy <- view $ aeTableSchemas.at tn
```

Comment:
we can delete this now - it makes sense that the bulk of this work is unwinding the tangle of partial pattern matches and making them complete, and making explicit use of something approximating `MonadFail#fail` instead.